src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
+
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_VALUE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZE_INDEX = 2;
+    private Node [] elements = new Node[DEFAULT_VALUE];
+    private int size;
+
+    private class Node<K,V> {
+        private Integer hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > elements.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHashCode(key) % elements.length;
+        if (elements[index] != null) {
+            Node currentBucket = elements[index];
+            do {
+                if (currentBucket.hash == getHashCode(key)
+                        && Objects.equals(key, currentBucket.key)) {
+                    currentBucket.value = value;
+                    return;
+                }
+                if (currentBucket.next != null) {
+                    currentBucket = currentBucket.next;
+                }
+            } while (currentBucket.next != null);
+            if (currentBucket.hash == getHashCode(key) && Objects.equals(key, currentBucket.key)) {
+                currentBucket.value = value;
+                return;
+            }
+            Node newNode = new Node<>(getHashCode(key), key, value, null);
+            currentBucket.next = newNode;
+            size++;
+            return;
+        }
+        elements[index] = new Node(getHashCode(key), key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getHashCode(key) % elements.length;
+        if (elements[index] != null) {
+            if (Objects.equals(key, elements[index].key)) {
+                return (V) elements[index].value;
+            } else if (elements[index].next != null) {
+                Node currentBucket = elements[index].next;
+                while (currentBucket != null) {
+                    if (Objects.equals(key, currentBucket.key)) {
+                        return (V) currentBucket.value;
+                    }
+                    currentBucket = currentBucket.next;
+                }
+            }
+        }
```

Comment:
too complex logic, let's simplify it:
1. Find bucket by index
2. use loop and go through all chain of this bucket, if keys equal the return value
3. if you went through chain and didn't find needed key return null in the end of this method

Reply:
Well done
60  +      int index = getHashCode(key) % elements.length;   - find bucket by index

The bucket may be null, isn't it? So let's check it:
61  +      if (elements[index] != null) {
              ...
              }   -  on the end of this block **return null**, if bucket is empty. This case return null if any other **return** (63 or 68) didn't work.

74  +     return null;

But if bucket is full,   "**2. use loop and go through all chain of this bucket, if keys equal the return value"**

If we use a **do...while** loop, we need check last next.node "by hand". I choose **while** loop, so I need check first next.node "by hand":
62  +    if (Objects.equals(key, elements[index].key)) {
63  +        return (V) elements[index].value;